hypothesis-python/tests/conjecture/common.py

```diff
@@ -40,6 +42,26 @@
 )
 
 
+def interesting_origin(n: Optional[int] = None) -> InterestingOrigin:
+    """
+    Creates and returns an InterestingOrigin, parameterized by n, such that
+    interesting_origin(n) == interesting_origin(m) iff n = m.
+    """
+    try:
+        int("not an int")
+    except Exception as e:
+        origin = InterestingOrigin.from_exception(e)
+        if n is None:
+            return origin
+        return InterestingOrigin(
+            exc_type=origin.exc_type,
+            filename=origin.filename,
+            lineno=n,
+            context=origin.context,
+            group_elems=origin.group_elems,
+        )
```

Comment:
Seems like this will collide for `n=None` and `n=<the lineno on which we create the error>`?  I think that's fine though, we don't make any promises about not-passing-`n`.

Also, a [nice feature of namedtuple](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace):

```suggestion
        return origin._replace(lineno=n)
```

Reply:
yeah, I realized this after, but every alternative for `n=None` collides with some passed value of `n`. We could modify a different field but this seems so unlikely to come up I didn't bother. (an invariant might be, "don't mix None and ints for n"). Added a comment